dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/UseCallingThreadExecutor.java

```diff
@@ -16,29 +16,58 @@
 
 package com.palantir.conjure.java.dialogue.serde;
 
+import com.codahale.metrics.Meter;
+import com.google.common.annotations.VisibleForTesting;
 import com.palantir.logsafe.Preconditions;
 import com.palantir.random.SafeThreadLocalRandom;
+import com.palantir.tritium.metrics.registry.SharedTaggedMetricRegistries;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.Random;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Package private API for enabling
  * {@link com.palantir.dialogue.blocking.CallingThreadExecutor} optimization.
  */
 final class UseCallingThreadExecutor {
 
+    private static final Logger log = LoggerFactory.getLogger(UseCallingThreadExecutor.class);
+    private static final float DEFAULT_PROBABILITY = 0.01f;
     private static final float DISABLED = 0.0f;
     private static final float ENABLED = 1.0f;
-    private static volatile float probability = DISABLED;
 
-    private UseCallingThreadExecutor() {}
+    /** All the calling code is static, does not make sense to plumb all that through for a temporary feature flag. */
+    @SuppressWarnings("deprecation")
+    private static final UseCallingThreadExecutor INSTANCE =
+            new UseCallingThreadExecutor(SafeThreadLocalRandom.get(), SharedTaggedMetricRegistries.getSingleton());
 
-    static boolean shouldUseCallingThreadExecutor() {
-        float currentProbability = probability;
-        if (currentProbability == DISABLED) {
-            return false;
-        } else if (currentProbability == ENABLED) {
-            return true;
-        }
-        return SafeThreadLocalRandom.get().nextFloat() < currentProbability;
+    private final Random random;
+    private final Consumer<RuntimeException> failureConsumer;
+    private final DialogueFeatureFlagsMetrics metrics;
+    private volatile float probability = DEFAULT_PROBABILITY;
+
+    @VisibleForTesting
+    UseCallingThreadExecutor(Random random, TaggedMetricRegistry taggedMetricRegistry) {
+        this.random = random;
+        this.metrics = DialogueFeatureFlagsMetrics.of(taggedMetricRegistry);
+        failureConsumer = runtimeException -> {
+            log.info("Unexpected exception from calling thread executor. Disabling this feature", runtimeException);
```

Comment:
I'd call this an error, wouldn't you?

I'm not sure how valuable this check is since it's not the most interesting way this could fail (for some definition of interesting), and I think there's a race condition where the caller is interrupted as a retry (or queued request) is submitted which results in RejectedExecutionException, but the system is still working as expected.

Reply:
aye, going to do as we discussed.